migrate_fields_config.py

```diff
@@ -0,0 +1,222 @@
+#! /usr/bin/python3
+"""
+Migrate field configuration from CloudV1 to CloudV2.
+To be used on a CloudV2 source having the same schema version as CloudV1.
+
+TODO: Move filtering of type date and sort=false sooner in the process
+"""
+
+from cloud_client import *
+import argparse
+
+# v1 -> v2
+FIELDS_KEYS_V1_V2 = {'fieldQueries': 'includeInQuery',
+                     'freeTextQueries': 'mergeWithLexicon',
+                     'facet': 'facet',
+                     'multivalueFacet': 'multiValueFacet',
+                     'sort': 'sort',
+                     'displayField': 'includeInResults'}
+
+KEY_V1_CONFIGNAME = 'v1ConfigName'
+KEY_V2_CONFIGNAME = 'v2ConfigName'
+KEY_V1_FIELD = 'v1Field'
+KEY_V2_FIELD = 'v2Field'
+KEY_V1_VALUE = 'v1Value'
+KEY_V2_VALUE = 'v2Value'
+KEY_CONFIG_DIFF = 'configDiff'
+
+
+def v1_get_source_id(sources, source_name):
+    """ Get a source id from a source name (CloudV1)
+    >>> v1_get_source_id([{'id': '0', 'name': 'FOO'}, {'id': '1', 'name': 'BAR'}], 'bar')
+    '1'
+    >>> v1_get_source_id([{'id': '0', 'name': 'FOO'}, {'id': '1', 'name': 'BAR'}], 'new')
+    Traceback (most recent call last):
+        ...
+    ValueError: Source new does not exist
+    >>> v1_get_source_id([{'id': '0', 'name': 'FOO'}, {'id': '1', 'name': 'FOO'}], 'foo')
+    Traceback (most recent call last):
+        ...
+    ValueError: More than one source foo found. This is wrong...
+    """
+    source_ids = [source['id'] for source in sources
+                  if source['name'].lower() == source_name.lower()]
+    if not source_ids:
+        raise ValueError(f'Source {source_name} does not exist')
+    if len(source_ids) > 1:
+        raise ValueError(f'More than one source {source_name} found. This is wrong...')
```

Comment:
This is wrong...
Clearer message perhaps?

Reply:
👍